src/components/spaces/permissions/editSpacePermissionsModal/UpdateWritePermission.tsx

```diff
@@ -0,0 +1,101 @@
+import { isEmptyArray } from '@subsocial/utils'
+import { useMemo } from 'react'
+import { useSubsocialApi } from 'src/components/substrate'
+import { TxButtonProps, TxCallback } from 'src/components/substrate/SubstrateTxButton'
+import TxButton from 'src/components/utils/TxButton'
+import { useCreateReloadSpace } from 'src/rtk/app/hooks'
+import { useAppDispatch } from 'src/rtk/app/store'
+import { useFetchSpaceEditors } from 'src/rtk/features/accounts/accountsHooks'
+import { upsertSpaceEditorsBySpaceId } from 'src/rtk/features/accounts/spaceEditorsSlice'
+import { SpaceStruct } from 'src/types'
+import { buildGrantOrRevokeRoleArgs } from '../../roles/editor-role'
+import { newWritePermission } from '../space-permissions'
+import useGetRoleId from '../useRoleCreated'
+import { BuiltInRole } from '../utils'
+
+type UpdateWritePermissionProps = TxButtonProps & {
+  onSuccess?: () => void
+  space?: SpaceStruct
+  whoCanPost: BuiltInRole
+  label?: string
+}
+
+export function EditWritePermission(props: UpdateWritePermissionProps) {
+  const { space, label, whoCanPost, onSuccess, ...buttonProps } = props
+  const { id } = space || {}
+  const spaceId = id || ''
+
+  console.log(space)
+  const reloadSpace = useCreateReloadSpace()
+  const { spaceEditors: editors = [], loading } = useFetchSpaceEditors(spaceId)
+  const { roleId, loaded } = useGetRoleId(spaceId)
+  const { api } = useSubsocialApi()
+  const dispatch = useAppDispatch()
+
+  const isHaveEditors = !isEmptyArray(editors)
+
+  const newTxParams = () => {
+    const update = {
+      permissions: newWritePermission(whoCanPost),
+    }
+
+    return [id, update]
+  }
+
+  const buildBatchTxParams = async () => {
+    if (!api || !roleId) return []
+
+    const batchTxs = [
+      api.tx.spaces.updateSpace(spaceId, {
+        permissions: newWritePermission(whoCanPost),
+      }),
+    ]
+
+    if (isHaveEditors) {
+      batchTxs.push(api.tx.roles.revokeRole(...buildGrantOrRevokeRoleArgs(roleId, editors)))
+    }
```

Comment:
I'm not sure if you need to remove the editors in this case

I think just removing them from the interface is enough

Because if I have a lot of editors, it will be crazy if I need to add them all again

Example of use: Subwallet has 10 editors (SMM department)

They decide to run a contest where all members of the community can write posts about SubWallet, they open up the space for followers, but want to close access after the contest and have just owner and the editors again